src/LanguageServer/Protocol/Features/Options/FallbackAnalyzerConfigOptionsProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Options;
+
+[Shared]
+[ExportWorkspaceService(typeof(IFallbackAnalyzerConfigOptionsProvider), workspaceKinds:
+    [WorkspaceKind.Host, WorkspaceKind.Interactive, WorkspaceKind.SemanticSearch, WorkspaceKind.MetadataAsSource, WorkspaceKind.MiscellaneousFiles, WorkspaceKind.Debugger, WorkspaceKind.Preview])]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class FallbackAnalyzerConfigOptionsProvider(EditorConfigOptionsEnumerator optionsEnumerator, IGlobalOptionService globalOptions) : IFallbackAnalyzerConfigOptionsProvider
```

Comment:
consider doc'ing.  even if it's just to say that it fallsback to loading from the global options service.

note: i would not name this FallbackAnalyzerConfigOptionsProvider.  I would prefer either FeaturesFallbackAnalyzerConfigOptionsProvider or GlobalOptionsFallbackAnalyzerConfigOptionsProvider

Reply:
naming it in this fasion seems like it's the lowest-level impl of the interface to me.

Reply:
Don't we usually name the lowest as `DefaultFallbackAnalyzerConfigOptionsProvider`? How about `ClientFallbackAnalyzerConfigOptionsProvider`? It's in LSP layer, which is for our client implementations (VS/VS Code).

Reply:
Client is perfect.  As i would associate global options with a client.  ('Host' also works fine as a reasonable name).

Reply:
> Don't we usually name the lowest as DefaultFallbackAnalyzerConfigOptionsProvider?

yesish.  I still have in my head that the name without the `I` and without any sort of interesting qualifier is like the defacto vesrion everyone should have.  THis impl here *is* special IMO, and warrants Client/Host/Editor or something in the name.  tnx.